File: longest-harmonious-subsequence.ts
{
    function findLHS(nums: number[]): number {
        const map = new Map();
        let max = 0;
    
        for (const num of nums) 
            map.set(num, (map.get(num) || 0) + 1);
    
        map.forEach((count, num) => {
            const next = map.get(num + 1);
    
            if (next)
                max = Math.max(max, count + next);
        })
    
        return max;
    };

    console.log(findLHS([-1,0,-1,0,-1,0,-1]))
}

File: longest-harmonious-subsequence.rs
use std::collections::HashMap;

pub fn find_lhs(nums: Vec<i32>) -> i32 {
    let mut map = HashMap::new();
    let mut max = 0;

    for &num in &nums {
        *map.entry(num).or_insert(0) += 1;
    }

    for (&num, &count) in &map {
        if let Some(&next) = map.get(&(num + 1)) {
            max = max.max(count + next);
        }
    }

    max
}
